player.py

```diff
@@ -3,41 +3,63 @@
 #player tank
 class Player(pygame.sprite.Sprite):
     def __init__(self, x, y, width, height):
-        #location
-        self.x = x
-        self.y = y
+        super().__init__()
+
+        # image = pygame.image.load(TANK_UP)
+        self.image = pygame.transform.scale(pygame.image.load(TANK_UP), (width, height))
+        self.direction = TANK_UP
+
         #size
         self.width = width
         self.height = height
+
+        #starting coordinates for the player 
+            #(not the player image)
+        self.x = x
+        self.y = y
+
+        #display
+        self.rect = self.image.get_rect()
+
+        #update image location
+        self.rect.x = self.x
+        self.rect.y = self.y
+        
         #scorekeep
         self.score = 0
 
-    #draw on screen
-    def main(self, display):
-        pygame.draw.rect(display, (255, 0, 0), (self.x, self.y, self.width, self.height))
-
     def update_player(self):
 
         #tank movement
+        #also updates the tank image to be facing in the direction of movement
         keys = pygame.key.get_pressed()
         if keys[pygame.K_a]:
-            self.x -= 2
-        if keys[pygame.K_d]:
-            self.x += 2
-        if keys[pygame.K_w]:
-            self.y-= 2
-        if keys[pygame.K_s]:
-            self.y+= 2
+            self.rect.x -= 2
+            self.image = pygame.transform.scale(pygame.image.load(TANK_LEFT), (self.width, self.height))
+            self.direction = TANK_LEFT
+        elif keys[pygame.K_d]:
+            self.rect.x += 2
+            self.image = pygame.transform.scale(pygame.image.load(TANK_RIGHT), (self.width, self.height))
+            self.direction = TANK_RIGHT
+        elif keys[pygame.K_w]:
+            self.rect.y-= 2
+            self.image = pygame.transform.scale(pygame.image.load(TANK_UP), (self.width, self.height))
+            self.direction = TANK_UP
+        elif keys[pygame.K_s]:
+            self.rect.y+= 2
+            self.image = pygame.transform.scale(pygame.image.load(TANK_DOWN), (self.width, self.height))
+            self.direction = TANK_DOWN
```

Comment:
Just a thought, but maybe pygame.transform.rotate() will be cleaner to use and require less image loads?

Reply:
So I tried to make the transform rotate to work but had a hard time with the conditional to make it only happen once while the key is being held. I had the tank spinning really fast in circles, lol it was pretty funny but also frustrating to figure out.